autopush/utils.py

```diff
@@ -108,13 +108,39 @@ def fix_padding(string):
     return string
 
 
+def decipher_public_key(key_data):
+    """A public key may come in several flavors. Attempt to extract the
+    valid key bits from keys doing minimal validation checks.
+
+    This is mostly a result of libs like WebCrypto prefixing data to "raw"
+    keys, and the ecdsa library not really providing helpful errors.
+
+    :param key_data: the raw-ish key we're going to try and process
+    :returns: the raw key data.
+    :raises: ValueError for unknown or poorly formatted keys.
+
+    """
+    # key data is actually a raw coordinate pair
+    key_len = len(key_data)
+    if key_len == 64:
+        return key_data
+    # Key format is "raw"
+    if key_len == 65 and key_data[0] == '\x04':
+        return key_data[-64:]
+    # key format is "spki"
+    if key_len == 88 and key_data[:3] == '0V0':
```

Comment:
@groovecoder 
`key_data[:3]` should return the first three characters, not the last three. 

The code allows for three types of "raw" (or raw-ish) key specification. All the keys have been translated from URL Save base64 before any validation is done. Base64 requires strings that are on long boundaries so some strings may require padding. fix_padding checks the string and adds sufficient padding characters ( '=' ) so that the string decodes properly.  

A true raw key, (which is 64 bytes long), a tagged raw key (which is 65 bytes) and a SPKI key, (which is 88 bytes and begins with a prefix that starts with `0V0`) . Honestly, I wouldn't worry about the SPKI format. 

The tagged raw key format is what is exported from window.crypto.subtle.exportKey() and was included because i believe node.js may use WebCrypto (or at least a variant of crypto.subtle), and there are some RFCs that stipulate that format for "raw" keys.
Wow, that was a lot of extra information, but hopefully helps clarify what you're seeing.
Now, as for what you're getting:
I'm not certain I know where self.vapid_key is being set or what value it's being set to. Normally, the VAPID key is contained in
`Crypto-Key: p256ecdsa=..."
(Fun fact, `Crypto-Key` can contain multiple sub keys, each is tagged with a unique key. `autopush.utils.parse_header` attempts to deal with this. The last element ( which you can access via [:-1]) should contain a dictionary of values you can use.)

If that is the value, I'd suggest injecting a `import pdb;pdb.set_trace()` and seeing what the values are and where `extract_public_key` falls over. 

(Oh, and a PR or issue if you find a bug in my code would be delightful.)

Does that help?